web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,444 @@
+ */
+
+import React, { useEffect } from "react";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core";
+
+import { _, N_ } from "~/i18n";
+import { deviceSize } from '~/components/storage/utils';
+import { If, OptionsPicker, Section, SectionSkeleton } from "~/components/core";
+import { noop, useLocalStorage } from "~/utils";
+import { sprintf } from "sprintf-js";
+import { Table, Thead, Tr, Th, Tbody, Td, TreeRowWrapper } from '@patternfly/react-table';
+
+/**
+ * @typedef {import ("~/client/storage").ProposalManager.ProposalSettings} ProposalSettings
+ * @typedef {import ("~/client/storage").ProposalManager.SpaceAction} SpaceAction
+ * @typedef {import ("~/client/storage").DevicesManager.StorageDevice} StorageDevice
+ */
+
+/**
+ * @typedef SpacePolicy
+ * @type {object}
+ * @property {string} id
+ * @property {string} label
+ * @property {string} description
+ */
+
+/** @type {SpacePolicy[]} */
+const SPACE_POLICIES = [
+  {
+    id: "delete",
+    label: N_("Delete current content"),
+    description: N_("All partitions will be removed and any data in the disks will be lost.")
+  },
+  {
+    id: "resize",
+    label: N_("Shrink existing partitions"),
+    description: N_("The data is kept, but the current partitions will be resized as needed.")
+  },
+  {
+    id: "keep",
+    label: N_("Use available space"),
+    description: N_("The data is kept. Only the space not assigned to any partition will be used.")
+  },
+  {
+    id: "custom",
+    label: N_("Custom"),
+    description: N_("Select what to do with each partition.")
+  }
+];
+
+// Names of the columns for the policy actions.
+const columnNames = {
+  device: N_("Used device"),
+  content: N_("Current content"),
+  size: N_("Size"),
+  details: N_("Details"),
+  action: N_("Action")
+};
+
+/**
+ * Column content with the description of a device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceDescriptionColumn = ({ device }) => {
+  return (
+    <>
+      <div>{device.name}</div>
+      <If
+        condition={device.isDrive}
+        then={<div className="fs-small">{`${device.vendor} ${device.model}`}</div>}
+      />
+    </>
+  );
+};
+
+/**
+ * Column content with information about the current content of the device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceContentColumn = ({ device }) => {
+  const PartitionTableContent = () => {
+    return (
+      <div>
+        {sprintf(_("%s partition table"), device.partitionTable.type.toUpperCase())}
+      </div>
+    );
+  };
+
+  const BlockContent = () => {
+    const renderContent = () => {
+      const systems = device.systems;
+      if (systems.length > 0) return systems.join(", ");
+
+      const filesystem = device.filesystem;
+      if (filesystem?.isEFI) return _("EFI");
+      if (filesystem) return sprintf(_("%s file system"), filesystem?.type);
```

Comment:
I am not sure if that "file system" suffix won't be confusing. E.g. "Btrfs file system" will mean "B tree file system file system" or ntfs or even nfs will be strange.

Reply:
I think it is ok "Btrfs file system" or "ntfs file system". At the end, we use "Btrfs", "Xfs", etc as a name.